plugin/src/main/java/com/denizenscript/denizen/events/block/DragonEggMovesScriptEvent.java

```diff
@@ -0,0 +1,66 @@
+package com.denizenscript.denizen.events.block;
+
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import org.bukkit.Material;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockFromToEvent;
+
+public class DragonEggMovesScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // dragon egg moves
+    //
+    // @Group Block
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a dragon egg moves.
+    //
+    // @Context
+    // <context.location> returns the LocationTag the egg started at.
+    // <context.destination> returns the LocationTag the egg teleported to.
+    //
+    // -->
+
+    public DragonEggMovesScriptEvent() {
+        registerCouldMatcher("dragon egg moves");
+    }
+
+    public LocationTag location;
+    public LocationTag destination;
+    public BlockFromToEvent event;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        if (!runInCheck(path, location) && !runInCheck(path, destination)) {
+            return false;
+        }
+        return super.matches(path);
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "location" -> location;
+            case "destination" -> destination;
+            default -> super.getContext(name);
```

Comment:
albeit pretty redundant, the `<context.material>` tag is technically being lost here - it should probably be left in the code for the odd chance of historical compat